### ISSUE TITLE
Updat value function with different action types, why?

### ISSUE THREAD

**Issue by tessavdheiden** [shariqiqbal2810/maddpg-pytorch#32]
Hi Shariq,

In your code you update the value function with actions computed by:
1) [gumbel_softmax](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L143)
2) [onehot_from_logits](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L153)

As far as I know, 1) has the gradient attached, while 2) does not. 

Why did you implemented it this way?

**Comment by uhlajs** [comment]
Hi @tessavdheiden,
I believe that in one call of [update](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L82) you want to update actor just for one actor (namely for actor attach to `agent_i`). That is the reason why you send one action sample with [gradient attached](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L150-L151) (representing the action of agent `agent_i`) and others [without](https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L152-L155).
